pkg/cloud/azure/services/groups/groups.go

```
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package groups
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+// Get provides Groups info to caller
+func (s *Service) Get(ctx context.Context) (interface{}, error) {
+	return s.Client.Get(ctx, s.Scope.ClusterConfig.ResourceGroup)
+}
+
+// CreateOrUpdate creates or updates Groups info
+func (s *Service) CreateOrUpdate(ctx context.Context) error {
+	_, err := s.Client.CreateOrUpdate(ctx, s.Scope.ClusterConfig.ResourceGroup, resources.Group{Location: to.StringPtr(s.Scope.ClusterConfig.Location)})
+	return err
+}
+
+// Delete deletes provided group name
+func (s *Service) Delete(ctx context.Context) error {
+	future, err := s.Client.Delete(ctx, s.Scope.ClusterConfig.ResourceGroup)
+	if err != nil {
+		return err
```

Let's return an error that wraps `err` here.